Another observation. If you change this in https://github.com/phpv8/v8js/blob/php7/v8js_convert.cc#L104

```
- long v;
+ zend_long v:
```

the warning for https://github.com/phpv8/v8js/blob/php7/v8js_convert.cc#L140 disappears. Instead a warning is issued on https://github.com/phpv8/v8js/blob/php7/v8js_convert.cc#L148:

`jsValue = V8JS_INT(v);
`

> ext\v8js\v8js_convert.cc(148): warning C4244: 'argument': conversion from 'zend_long' to 'int32_t', possible loss of data

Interpretation: v is zend_long, V8JS_INT(v) is defined as 'v8::Integer::New(isolate, v)', jsValue is just a (typeless?) v8::Handle<v8::Value>. Something tells me that V8JS_INT(v) should have been int64_t and not int32_t.

--- a/tests/long_int64.phpt
+++ b/tests/long_int64.phpt
@@ -1,7 +1,11 @@
 --TEST--
 Test V8::executeString() : Check long 64-bit export from PHP to JS
 --SKIPIF--
-<?php require_once(dirname(__FILE__) . '/skipif.inc'); ?>
+<?php
+require_once(dirname(__FILE__) . '/skipif.inc');
+if (4 == PHP_INT_SIZE)
+	die("SKIP test not supported on 32bit PHP");
+?>
 --FILE--
 <?php
 

--- a/tests/long_int64.phpt
+++ b/tests/long_int64.phpt
@@ -1,0 +1,32 @@
+--TEST--
+Test V8::executeString() : Check long 64-bit export from PHP to JS
+--SKIPIF--
+<?php require_once(dirname(__FILE__) . '/skipif.inc'); ?>
+--FILE--
+<?php
+
+$addInt = function ($a, $b) {
+    return $a + $b;
+};
+
+$v8 = new V8Js('PHP');
+$v8->add = $addInt;
+$v8->p1 = 10;
+$v8->p2 = pow(2,45);
+
+var_dump($v8->p2);
+
+var_dump($v8->executeString('
+	var res = PHP.p1 + PHP.p2;
+	"p1:" + PHP.p1 +
+	", p2:" + PHP.p2 +
+	", PHP.add(p1,p2)=" + PHP.add(PHP.p1, PHP.p2) +
+	", p1+p2=" + res +
+	" -> " + (new Date(res)).toISOString();
+	'));
+?>
+===EOF===
+--EXPECT--
+int(35184372088832)
+string(105) "p1:10, p2:35184372088832, PHP.add(p1,p2)=35184372088842, p1+p2=35184372088842 -> 3084-12-12T12:41:28.842Z"
+===EOF===

--- a/v8js_convert.cc
+++ b/v8js_convert.cc
@@ -101,7 +101,7 @@ static v8::Handle<v8::Value> v8js_hash_to_jsarr(zval *value, v8::Isolate *isolat
 v8::Handle<v8::Value> zval_to_v8js(zval *value, v8::Isolate *isolate TSRMLS_DC) /* {{{ */
 {
 	v8::Handle<v8::Value> jsValue;
-	long v;
+	zend_long v;
 	zend_class_entry *ce;
 
 	switch (Z_TYPE_P(value))
@@ -143,9 +143,9 @@ v8::Handle<v8::Value> zval_to_v8js(zval *value, v8::Isolate *isolate TSRMLS_DC) 
 #undef max
 #undef min
 			if (v < - std::numeric_limits<int32_t>::min() || v > std::numeric_limits<int32_t>::max()) {
-				jsValue = V8JS_FLOAT((double)v);
+				jsValue = V8JS_FLOAT(static_cast<double>(v));
 			} else {
-				jsValue = V8JS_INT(v);
+				jsValue = V8JS_INT(static_cast<int32_t>(v));
 			}
 			break;
 

